File: app/src/main/java/com/pietrantuono/podcasts/addpodcast/model/AddPodcastsModel.java
package com.pietrantuono.podcasts.addpodcast.model;


import io.reactivex.observers.DisposableObserver;

public interface AddPodcastsModel {
    void subscribeToSearch(DisposableObserver<SearchResult> observer);

    void unsubscribeFromSearch();

    void searchPodcasts(String query);

}

File: app/src/main/java/com/pietrantuono/podcasts/addpodcast/model/AddPodcastsModel.kt
package com.pietrantuono.podcasts.addpodcast.model


import com.pietrantuono.podcasts.addpodcast.singlepodcast.view.State
import rx.Observer

interface AddPodcastsModel {
    var viewState: State?

    var query: String?

    fun subscribeToSearch(observer: Observer<SearchResult>)

    fun unsubscribeFromSearch()

    fun searchPodcasts(query: String?)

    fun refresh()
}
